File: ticket-frontend/src/modules/event/dtos/event-dto.ts
import { LabelValue } from '../../core/label-value';
import { TypeUtils } from '../../core/type-utils';
import { MovieListItemDto } from '../../movie/dtos/movie-list-item-dto';
import { TicketDto } from '../../ticket/dtos/ticket-dto';

export class EventDto {
    eventId: number | null;
    name: string;
    eventDatetime: Date | null;
    description: string;
    eventPrice: number;
    movieRef: MovieListItemDto | null;
    hallRef: LabelValue<number> | null;
    tickets: TicketDto[];

    constructor(obj: {
        eventId?: number | null | undefined,
        name: string,
        eventDatetime?: Date | null | undefined,
        description: string,
        eventPrice: number,
        movieRef?: MovieListItemDto | null | undefined,
        hallRef?: LabelValue<number> | null | undefined,
        tickets: TicketDto[],
    }) {
        this.eventId = obj.eventId ? obj.eventId : null;
        this.name = obj.name;
        this.eventDatetime = obj.eventDatetime ? obj.eventDatetime : null;
        this.description = obj.description;
        this.eventPrice = obj.eventPrice;
        this.movieRef = obj.movieRef ? obj.movieRef : null;
        this.hallRef = obj.hallRef ? obj.hallRef : null;
        this.tickets = obj.tickets;
    }

    static fromObj(obj: any): EventDto | null {
        if (!obj) {
            return null;
        }
        const eventDto: EventDto = new EventDto({
            ...obj,
            eventDatetime: obj.eventDatetime ? new Date(obj.eventDatetime) : null,
            movieRef: MovieListItemDto.fromObj(obj.movieRef),
            hallRef: LabelValue.fromObj(obj.hallRef),
            tickets: TicketDto.listFromObjList(obj.tickets),
        });
        return eventDto;
    }

    static listFromObjList(objs: any[]): EventDto[] {
        if (!objs) {
            return [];
        }
        return objs.map(EventDto.fromObj).filter(TypeUtils.isNonNullable);
    }
}

File: ticket-backend/src/main/java/aics/domain/event/dtos/EventDto.java
package aics.domain.event.dtos;

import aics.domain.event.entities.Event;
import aics.domain.movie.dtos.MovieListItemDto;
import aics.domain.ticket.dtos.TicketDto;
import aics.infrastructure.core.LabelValue;
import lombok.Data;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.List;
import java.util.stream.Collectors;

@Data
@Accessors(chain = true)
public class EventDto implements Serializable {
    private Long eventId;
    private String name;
    private LocalDateTime eventDatetime;
    private String description;
    private Double eventPrice;
    private MovieListItemDto movieRef;
    private LabelValue<Long> hallRef;
    private List<TicketDto> tickets;

    public static EventDto fromEvent(Event event) {
        if (event == null) {
            return null;
        }
        LabelValue<Long> hallRef = event.getHall() != null
            ? new LabelValue<>(event.getHall().getProvider().getName() + "-" + event.getHall().getName(), event.getHall().getHallId())
            : null;
        return new EventDto()
            .setEventId(event.getEventId())
            .setName(event.getName())
            .setEventDatetime(event.getEventDatetime())
            .setDescription(event.getDescription())
            .setEventPrice(event.getEventPrice())
            .setMovieRef(MovieListItemDto.fromMovie(event.getMovie()))
            .setHallRef(hallRef)
            .setTickets(event.getTickets().stream().map(TicketDto::fromTicket).collect(Collectors.toList()));
    }
}
